relay-server/src/metrics_extraction/transactions.rs

```diff
@@ -51,20 +49,31 @@ pub struct CustomMeasurementConfig {
     limit: usize,
 }
 
+/// Maximum supported version of metrics extraction from transactions.
+///
+/// The version is an integer scalar, incremented by one on each new version.
+const EXTRACT_MAX_VERSION: u16 = 1;
+
 /// Configuration for extracting metrics from transaction payloads.
 #[derive(Default, Debug, Clone, Serialize, Deserialize)]
 #[serde(default, rename_all = "camelCase")]
 pub struct TransactionMetricsConfig {
+    /// The required version to extract transaction metrics.
+    version: u16,
     extract_metrics: BTreeSet<String>,
     extract_custom_tags: BTreeSet<String>,
     satisfaction_thresholds: Option<SatisfactionConfig>,
     custom_measurements: CustomMeasurementConfig,
 }
 
-#[cfg(feature = "processing")]
+impl TransactionMetricsConfig {
+    pub fn is_enabled(&self) -> bool {
+        self.version > 0 && self.version <= EXTRACT_MAX_VERSION
```

Comment:
Quick thought: We could actually remove the `self.version > 0` check and set `EXTRACT_MAX_VERSION = 0`. That way, we would not need any changes on the sentry side. Or am I missing something crucial?

Reply:
Actually, let's keep it as-is to be consistent with [session metrics extraction](https://github.com/getsentry/relay/blob/0576be7d503f8155e58cda1492216ab64631b05e/relay-server/src/metrics_extraction/sessions.rs#L35).